Tests/MapboxMapsTests/Foundation/Mocks/MockNotificationCenter.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+@testable import MapboxMaps
+
+final class MockNotificationCenter: NotificationCenterProtocol {
```

Comment:
Is it worth it to mock notification center? NotificationCenter.default delivers notifications synchronously already, and if we verify that the expected behaviors happen when a notification is posted, we'll know that the observer was added correctly.

Reply:
I think the advantage of Mocking Notification Center is in that allows us to directly verify when subscription/unsubscription happens, instead of relying on indirects signs.